script/bench.rb

```diff
@@ -182,17 +184,21 @@ def port_available?(port)
 puts "Populating Profile DB"
 run("bundle exec ruby script/profile_db_generator.rb")
 
-puts "Getting api key"
-api_key = `bundle exec rake api_key:create_master[bench]`.split("\n")[-1]
+puts "Getting admin api key"
+admin_api_key = `bundle exec rake api_key:create_master[bench]`.split("\n")[-1]
 
-def bench(path, name)
+puts "Getting user api key"
+user_api_key = `bundle exec rake user_api_key:create[user1]`.split("\n")[-1]
+
+def bench(path, name, headers)
   puts "Running apache bench warmup"
   add = ""
   add = "-c #{@concurrency} " if @concurrency > 1
-  `ab #{add} -n 20 -l "http://127.0.0.1:#{@port}#{path}"`
+  header_string = headers&.map { |k, v| "-H \"#{k}:#{v}\"" }&.join(" ")
+  `ab #{add} #{header_string} -n 20 -l "http://127.0.0.1:#{@port}#{path}"`
```

Comment:
Previous benchmark did not include the headers which means we were always running benchmarks as an anon user.